dramatiq/actor.py

```diff
@@ -85,6 +85,8 @@ def message_with_options(self, *, args=None, kwargs=None, **options):
         """
         for name in ["on_failure", "on_success"]:
             callback = options.get(name)
+            if callback is None:
```

Comment:
Did you intend to keep this change here?

Reply:
It's a minor optimization that skips the call overhead of the `if...elif`. IIRC it was ~30% of the saved function calls. Granted, it will will add overhead for cases were both callbacks are supplied, which I assumed is rarely the case.

Your call if you want to keep it, I'll update the PR accordingly :)

Reply:
Let's drop it. I don't imagine it made up 30% of the total time so it's probably not worth optimizing compared to `declare_queue` which involves IO.

Reply:
I pulled the profile code from my remote machine (note to self, sync scratch files):

2704297 function calls (without patch)
2684297 function calls (with patch but without continue)
2644297 function calls (with patch and continue)

So it's actually 2/3 of the saved calls, oof. Let me check the overhead for having one or both callbacks defined, brb.

Reply:
# one callback defined (method)
2644297 calls (with patch)
2684297 calls (with patch but without continue)

# both callbacks defined (method)
2644297 calls (with patch)
2684297 calls (with patch but without continue)

Interesting, looks like the compiler is already doing some optimizations, so no overhead for the additional continue conditional when the callbacks are not None.

Reply:
Yes, I can imagine there would be many calls here, but calls by themselves don't concern me that much. What are the timings like? My guess is the vast majority of the time is being spent in the declare_queue calls rather than here.

Reply:
It's 6ms/0.4% (1.1% instead of 1.5%) for 10k tasks. I've attached the call graphs for both version ((offtopic to this PR, switching to [orjson](https://github.com/ijl/orjson) might offer some nice gains)) :

![Screenshot_20200606_202458_without_continue](https://user-images.githubusercontent.com/1685335/83951792-6f762180-a834-11ea-8f48-42e3d5b19d07.png)
![Screenshot_20200606_202545_with_continue](https://user-images.githubusercontent.com/1685335/83951797-70a74e80-a834-11ea-9173-52c6b231b2d5.png)